From dec0608369e1fe50685ce94113681ec9ed1bc43c Mon Sep 17 00:00:00 2001
From: David Redondo <kde@david-redondo.de>
Date: Tue, 13 Aug 2024 09:09:06 +0000
Subject: [PATCH] xembedsniproxy: Guard against crash when xcb_query_tree_reply
 is null

BUG:491509
FIXED-IN:6.1.5


(cherry picked from commit 85eb56e174bf887ccb8311e8f1a871e1ec453e04)

Co-authored-by: David Redondo <kde@david-redondo.de>
---
 xembed-sni-proxy/sniproxy.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/xembed-sni-proxy/sniproxy.cpp b/xembed-sni-proxy/sniproxy.cpp
index 41e1fb0550c..21d74c566d5 100644
--- a/xembed-sni-proxy/sniproxy.cpp
+++ b/xembed-sni-proxy/sniproxy.cpp
@@ -76,6 +76,9 @@ static bool checkWindowOrDescendantWantButtonEvents(xcb_window_t window)
     }
     auto treeCookie = xcb_query_tree(connection, window);
     UniqueCPointer<xcb_query_tree_reply_t> tree(xcb_query_tree_reply(connection, treeCookie, nullptr));
+    if (!tree) {
+        return false;
+    }
     std::span<xcb_window_t> children(xcb_query_tree_children(tree.get()), xcb_query_tree_children_length(tree.get()));
     return std::ranges::any_of(children, &checkWindowOrDescendantWantButtonEvents);
 }
-- 
GitLab

